3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,25 +8,23 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
```

Comment:
In contrast to the rolling dice in the later exercises which were supposed to roll on, in this case it is best to exit this function once the promise is rejected as it has no work to do. Consider it luck that this function does not fail if you leave out the return 🙂.


```suggestion
        reject(new Error("You didn't pass in a first name!"));
        return;
```